packages/components/src/text-input/hooks/use-text-input.js

```diff
@@ -0,0 +1,194 @@
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useContextSystem } from '../../ui/context';
+import { useBaseField } from '../../base-field';
+import { useFormGroupContextId } from '../../ui/form-group';
+import * as styles from '../styles';
+import { useTextInputState } from './use-text-input-state';
+
+const useRootEventHandlers = ( {
+	dragHandlersRef,
+	inputRef,
+	isFocused,
+	isTypeNumeric,
+} ) => {
+	const canScroll = isTypeNumeric && isFocused;
+	const dragHandlers = dragHandlersRef.current;
+
+	useEffect( () => {
+		const handleOnWheel = () => {};
+		if ( inputRef.current.addEventListener ) {
+			inputRef.current.addEventListener( 'wheel', handleOnWheel, {
+				passive: false,
+			} );
+		}
+	}, [ inputRef, canScroll ] );
+
+	const handleOnClick = useCallback( () => {
+		inputRef.current.focus();
+	}, [ inputRef ] );
+
+	const handleOnTouchStart = useCallback( () => {
+		inputRef.current.focus();
+	}, [ inputRef ] );
+
+	return {
+		...dragHandlers,
+		onClick: handleOnClick,
+		onTouchStart: handleOnTouchStart,
+	};
+};
+
+/**
+ *
+ * @param {import('@wp-g2/create-styles').ViewOwnProps<import('../types').Props, 'input'>} props
+ */
+export function useTextInput( props ) {
+	const {
+		align,
+		arrows = true,
+		className,
+		defaultValue = '',
+		disabled,
+		dragAxis,
+		error = false,
+		format,
+		gap = 2.5,
+		id: idProp,
+		incrementFromNonNumericValue = false,
+		isCommitOnBlurOrEnter = false,
+		isFocused: isFocusedProp,
+		isInline = false,
+		isResizable = false,
+		isShiftStepEnabled = true,
+		justify,
+		max,
+		min,
+		multiline = false,
+		prefix,
+		shiftStep = 10,
+		size = 'medium',
+		step,
+		suffix,
+		type = 'text',
+		validate,
+		value: valueProp,
+		...otherProps
+	} = useContextSystem( props, 'TextInput' );
+
+	const id = useFormGroupContextId( idProp );
+
+	const {
+		decrement,
+		dragHandlersRef,
+		increment,
+		inputRef,
+		isFocused,
+		isInputTypeNumeric,
+		isTypeNumeric,
+		scrollHandlers,
+		value,
+		...textInputState
+	} = useTextInputState( {
+		...otherProps,
+		format,
+		defaultValue,
+		incrementFromNonNumericValue,
+		isCommitOnBlurOrEnter,
+		isFocused: isFocusedProp,
+		isShiftStepEnabled,
+		max,
+		min,
+		shiftStep,
+		step,
+		value: valueProp,
+		validate,
+		type,
+	} );
+
+	const rootEventHandlers = useRootEventHandlers( {
+		inputRef,
+		dragHandlersRef,
+		isTypeNumeric,
+		isFocused,
+	} );
+
+	const baseFieldProps = useBaseField( {
+		align,
+		disabled,
+		error,
+		gap,
+		isFocused,
+		isInline,
+		justify,
+	} );
+
+	const InputComponent = multiline ? TextareaAutosize : 'input';
+
+	const classes = useMemo(
+		() =>
+			cx(
+				baseFieldProps.className,
+				multiline && styles.multiline,
+				className
+			),
+		[ baseFieldProps.className, className, multiline ]
+	);
+
+	const inputClasses = useMemo(
+		() =>
+			cx(
+				styles.Input,
```

Comment:
Storybook is having trouble displaying this component. 

When it works, it borks around here. I'm not entirely sure why. 

Have I imported the styles correctly?

```bash
index.js:54 ReferenceError: Cannot access 'Input' before initialization
    at Module.Input (main.1bb8fefae19da236e3d5.hot-update.js:17)
    at use-text-input.js:150
    at mountMemo (react-dom.development.js:15442)
    at Object.useMemo (react-dom.development.js:15738)
    at useMemo (react.development.js:1521)
    at useTextInput (use-text-input.js:150)
    at TextInput (text-input.js:22)
    at renderWithHooks (react-dom.development.js:14803)
    at updateForwardRef (react-dom.development.js:16816)
    at beginWork (react-dom.development.js:18645)
```